src/Table/Table.tsx

```diff
@@ -1,8 +1,21 @@
-/** @jsx jsx */
-import { jsx } from "@emotion/core";
+import { ClassNames } from "@emotion/core";
 import React from "react";
 import * as typography from "../typography";
 import { colors } from "../colors";
+import { assertUnreachable } from "../shared/assertUnreachable";
+
+type As = React.ReactElement | keyof JSX.IntrinsicElements;
+
+/**
+ * Take an `as` value and make it into a `React.ReactElement`
+ */
+function createElementFromAs(as: As): React.ReactElement {
+  return React.isValidElement(as)
+    ? as
+    : typeof as === "string"
+    ? React.createElement(as)
+    : assertUnreachable(as);
+}
```

Comment:
NBD 
Was trying to think of a way for this to also handle the object as
```
const bodyTrElement = React.isValidElement(trAs)
    ? trAs
    : typeof trAs === "string"
    ? React.createElement(trAs)
    : createElementFromAs(trAs.body || "tr");
```
Something like this
```suggestion
function createElementFromAs<Key extends string>(
  as: As | { [key in Key]?: As } | undefined,
  defaultValue: As,
  key?: Key
): React.ReactElement {
  return React.isValidElement(as)
    ? as
    : typeof as === "string"
    ? React.createElement(as)
    : createElementFromAs((key && as?.[key]) ?? defaultValue, defaultValue);
}
```
would let use change that to 
```
const bodyTrElement = createElementFromAs(trAs, "tr", "body");
```

Reply:
Thanks for this suggestion! I'm happy the conversation was hot to better handle the object or value instead of should we even have an object or a value 😃 I'm not sure what the right abstraction here would be yet; and I would love to have one because I'd much rather inline the element generation. I'm going to leave this as it is for now and keep thinking about it 😃 

Something else I didn't add, but we might want in the future, is another option in `As` to make the element a function of the row props for the `tr`, the field props for `td`, and the column config (?) for the `th`. Def a use case we can figure out when we need it!